module/LineTraceArea.cpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file LineTraceArea.cpp
+ * @brief ライントレースエリアを攻略するクラス
+ * @author Hisataka-Hagiyama,uchyam
+ */
+
+#include "LineTraceArea.h"
+
+// Lコースの情報を初期化する
+const std::array<SectionParam, LineTraceArea::LEFT_SECTION_SIZE> LineTraceArea::LEFT_COURSE_INFO
+    = { SectionParam{ 1625.340, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 594.506, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 690.288, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 511.724, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 1709.700, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 309.562, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 807.404, 50, 100, PidGain(0, 0, 0) } };
+
+// Rコースの情報を初期化する
+const std::array<SectionParam, LineTraceArea::RIGHT_SECTION_SIZE> LineTraceArea::RIGHT_COURSE_INFO
+    = { SectionParam{ 1625.340, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 594.506, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 690.288, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 511.724, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 1709.700, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 309.562, 50, 100, PidGain(0, 0, 0) },
+        SectionParam{ 807.404, 50, 100, PidGain(0, 0, 0) } };
+
+void LineTraceArea::runLineTraceArea()
+{
+  const SectionParam* param;
+  bool isLeftEdge;  // true:左エッジ,false:右エッジ
+
+  if(IS_LEFT_COURSE) {
+    // Lコースの場合
+    param = LEFT_COURSE_INFO.begin();
+    isLeftEdge = false;
+  } else {
+    // Rコースの場合
+    param = RIGHT_COURSE_INFO.begin();
+    isLeftEdge = true;
+  }
```

Comment:
好みの問題なので、修正するかどうかは任せます...

個人的には、
```suggestion
  if(IS_LEFT_COURSE) {
    param = LEFT_COURSE_INFO.begin();
  } else {
    param = RIGHT_COURSE_INFO.begin();
  }
  isLeftEdge = !IS_LEFT_COURSE;
```
のほうが直感的?な気がして好きです。

Reply:
Lコースだったら右エッジでRコースだったら左エッジって絶対だっけ？
そうじゃないなら今のままでも良さそう

Reply:
ライントレースエリアではエッジは必ずコースの逆になるんじゃない？
だから上田君の言ってる反転させる代入文の方が個人的にはいいと思うけど，これはこれで理解はしやすいのかもしれないから好みの問題だね